src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskForSearchCompletionControllerTest.java

```diff
@@ -50,6 +50,38 @@ public void setUp() {
         common.clearAllRoleAssignments(authenticationHeaders);
     }
 
+    @Test
+    public void given_processApplication_task_when_decideAnApplication_event_then_return_processApplication_tasks() {
+        TestVariables processApplicationTaskVariables = common.setupTaskAndRetrieveIdsWithCustomVariablesOverride(
+            Map.of(
+                CamundaVariableDefinition.TASK_TYPE, "processApplication",
+                CamundaVariableDefinition.TASK_ID, "processApplication"
+            ));
+
+        SearchEventAndCase decideAnApplicationSearchRequest = new SearchEventAndCase(
+            processApplicationTaskVariables.getCaseId(),
+            "decideAnApplication",
+            "IA",
+            "Asylum"
+        );
+
+        common.setupOrganisationalRoleAssignment(authenticationHeaders);
+
+        Response result = restApiActions.post(
+            ENDPOINT_BEING_TESTED,
+            decideAnApplicationSearchRequest,
+            authenticationHeaders
+        );
+
+        result.then().assertThat()
+            .statusCode(HttpStatus.OK.value())
+            .contentType(APPLICATION_JSON_VALUE)
+            .body("tasks.size()", equalTo(1))
+            .body("tasks[0].type", equalTo("processApplication"));
+
```

Comment:
Can we add the test to capture the scenario where there is no match and an empty list is returned?